test/Fencer/Server/Test.hs

```diff
@@ -85,6 +87,49 @@ test_serverResponseNoRules =
       , Proto.rateLimitRequestHitsAddend = 0
       }
 
+-- | Test that when Fencer is started and there are no rules after
+-- loading (i.e., 'reloadRules' has been ran, but there is no
+-- configuration), requests to Fencer will be responded to with OK.
+--
+-- This behavior matches @lyft/ratelimit@.
```

Comment:
I think that here we also should test several requests in a row. And the common logic between `test_serverResponseNoRules` and `test_serverResponseNoConfigurationDirectory` should be abstracted out, if possible.

Reply:
I turned the `test_serverResponseNoRules` into a request sequence test.

Regarding `test_serverResponseNoConfigurationDirectory`, I'll see what can be done, though at the moment these two tests live in unrelated branches.

Reply:
> And the common logic between `test_serverResponseNoRules` and `test_serverResponseNoConfigurationDirectory` should be abstracted out, if possible

I just looked at this and in my opinion there is no benefit in trying to combine the two by abstracting common parts because there are not as many common parts. Although it might appear at first that these are almost identical test cases, they test two subtly different things: failing to find and load any rules and loading an empty rule list; the former does not set a flag in the application state unlike the other. This affects what is and how tested.

I might be missing to spot an elegant abstraction that makes the two equal at one level, but so far I haven't been able to see it.

Reply:
No abstraction is certainly better than a leaking abstraction.